keda/templates/manager/clusterrole.yaml

```diff
@@ -26,77 +30,59 @@ rules:
   - events
   verbs:
   - '*'
+{{- if not .Values.permissions.operator.restrict.secret }}
 - apiGroups:
   - ""
-  resources:
-  - external
-  - pods
-    {{- if eq .Values.permissions.operator.restrict.secret false }}
+  resources:    
   - secrets
-    {{- end }}
-  - services
-  verbs:
-  - get
-  - list
-  - watch
-- apiGroups:
-  - ""
-  resources:
-  - limitranges
   verbs:
   - list
   - watch
+  {{- with .Values.permissions.operator.restrict.namesAllowList }}
 - apiGroups:
   - ""
   resources:
-  - serviceaccounts
-  verbs:
-  - list
-  - watch
-- apiGroups:
-  - '*'
-  resources:
-  - '*'
+  - secrets
   verbs:
   - get
+  resourceNames: {{ toYaml . | nindent 4 }}
+  {{- end }}
+{{- end }}
+
+{{- if .Values.rbac.enabledCustomScaledRefKinds }}
+{{- if not .Values.rbac.scaledRefKinds }}
 - apiGroups:
   - '*'
   resources:
   - '*/scale'
   verbs:
-  - get
-  - list
-  - patch
-  - update
-  - watch
```

Comment:
I think that we can just keep the previous verbs as they blocked the deletions an * grants the removal permission too

Reply:
you are right, I didn't realise this. I am adding it back so that deletes are not possible. Good catch.